### ISSUE TITLE
Ranking order is lost when survey is reloaded

### ISSUE THREAD

**Issue by digiscapeuk** [surveyjs/survey-library#8836]
Consider a survey such as the attached:
[ranking bug.json](https://github.com/user-attachments/files/17073358/ranking.bug.json)

I've simplified the use case to illustrate the issue.

We have a set of user entered values - in this case it is there goals.  The user could enter up to 3 but it's optional whether they fill all of them out.

Once the user has entered their goals we'd like a later step to be ranking them in order of priority.

We can set the options in the ranking question by pulling the text from the earlier question and hide options that would be blank.

Ranking the options works fine and is stored correctly in the JSON on save.

The problem we are seeing is that when the survey is reloaded the ranking is lost and reverts to the initial sort that matches the order of items in the multipletext question.

I would expect it populate the choices using the earlier question but reapply the ranking using the data that was loaded from the json we have stored.

Documentation talks about feeding ranking questions with choices from other questions but that isn't appropriate for this use case.

**Comment by JaneSjs** [comment]
Hello @digiscapeuk,
Thank you for sharing your usage scenario. I reproduced the issue on my end ([View Demo](https://plnkr.co/edit/c0bTTmti9IdDsZlI)). Regardless of the changed sequence of Ranking items, they appear in their initial order:
```
survey.data = {
        "goals": {
            "goal1": "11",
            "goal3": "33"
        },
        "rankedGoals": [
            "3",
            "1"
        ]
};
```
![image](https://github.com/user-attachments/assets/cb513463-81e3-4d20-af68-4abc474f922c)

We'll research this issue and let you know our results. Please stay tuned.

**Comment by digiscapeuk** [comment]
That great. Thanks for the quick turnaround.Sent from my iPhoneOn 1 Oct 2024, at 08:42, Jane ***@***.***> wrote:﻿
Hello @digiscapeuk,
We have fixed this issue. The fix will be available with the upcoming maintenance update v1.12.4.
